src/Platform/AcquiaCloudPlatform.php

```diff
@@ -304,8 +331,65 @@ public function getActiveDomains(): array {
    */
   public function prefixDomain(string $domain, string $env_id): string {
     $http_conf = $this->get(self::ACE_SITE_HTTP_PROTOCOL);
-    $prefix = isset($http_conf[$env_id]) ? $http_conf[$env_id] : 'https://';
+    $prefix = $http_conf[$env_id] ?? 'https://';
     return $prefix . $domain;
   }
 
+  /**
+   * Fetch the location of the platform group file.
+   *
+   * @return string
+   *   Grouping file path.
+   */
+  protected function groupingSitesFilePath(): string {
+    $alias = $this->getAlias();
+    $dir_parts = static::PLATFORM_LOCATION;
+    array_unshift($dir_parts, getenv('HOME'));
+    $file = implode(DIRECTORY_SEPARATOR, $dir_parts) . "/{$alias}-groups.yml";
+
+    return $file;
+  }
+
+  /**
+   * Filter list of sites via group sites.
+   *
+   * @param string $group_name
+   *   Platform grouping sites placeholder.
+   * @param array $sites
+   *   Platform sites.
+   * @param OutputInterface $output
+   *   Output stream.
+   *
+   * @return array|int
+   *   Array containing filtered list of sites.
+   */
+  protected function filterSitesByGroup(string $group_name, array $sites, OutputInterface $output) {
+    $group_file = $this->groupingSitesFilePath();
+    try {
+      $file = Yaml::parseFile($group_file);
+    }
+    catch (ParseException $exception) {
+      $output->writeln('<error>Unable to parse the YAML ' . $exception->getMessage() . '</error>', );
+      return 1;
+    }
+
+    if (!isset($file[$group_name])) {
+      $output->writeln('<error>Group name doesn\'t exists.</error>');
+      return 2;
+    }
+    elseif (empty($file[$group_name])) {
+      $output->writeln('<warning>No sites available in the groups. Exiting...</warning>');
+      return 3;
+    }
+
+    foreach ($sites as $key => $site) {
```

Comment:
Correct me if I am wrong, but would not be `array_intersect_key()` function could do this? It is possible I misunderstanding the goal here.